packages/camera/camera_avfoundation/ios/Classes/CameraPlugin.m

```diff
@@ -104,8 +106,12 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
   });
 }
 
-- (void)handleMethodCallAsync:(FlutterMethodCall *)call result:(FlutterResult)result {
-  if ([@"availableCameras" isEqualToString:call.method]) {
+- (void)availableCamerasWithCompletion:
+    (nonnull void (^)(NSArray<FCPPlatformCameraDescription *> *_Nullable,
+                      FlutterError *_Nullable))completion {
+  // This doesn't interact with FLTCam, so can use an arbitrary thread rather than
+  // captureSessionQueue. It should still not be done on the main thread, however.
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
We will unfortunately need to have the thread dispatch in every individual method in the Pigeon version (instead of handleMethodCall doing it for everything at once). Almost everything needs to use `captureSessionQueue`, since `FLTCam` sometimes needs to redispatch to that queue in internal callbacks to interact with instance state, which means we can't just use task queues.

(Also IIRC task queues aren't implemented for macOS, and we want to enable this plugin for macOS at some point.)

It'll just be a couple of lines in each method though, so it shouldn't be too bad. I think we can simplify when we migrate to Swift.

Reply:
I think it's still better to use `captureSessionQueue` for all session configurations, in case Apple made any assumptions in its internal implementation (e.g. I wouldn't be surprised if multiple capture session related API access some common internal states).

(`AVCaptureDeviceDiscoverySession`'s API doc doesn't say anything, but Apple's [sample project](https://developer.apple.com/documentation/avfoundation/capture_setup/avcam_building_a_camera_app) uses this convention)

Reply:
Done. It hadn't occurred to me that in theory the unique device IDs could be thread-specific; that would be weird, but not implausible.

Reply:
Ya I bet we should be fine either way, but just wanna be extra safe here.